interfaces/builtin/README.md

```diff
@@ -0,0 +1,308 @@
+# Interfaces static information
+
+## BaseDeclaration{Plugs,Slots}
+
+The snap-declarations are mainly used to control what plugs or slots a snap is
```

Comment:
it's  a bit strange that this mentions snap-declaration but not base declaration which is the thing actually controlled here. We need to expand this to cover a bit more. The base declaration it's snapd base rules and then snap-declaration can be used to override those.